meeteval/io/base.py

```diff
@@ -20,12 +23,33 @@
     Subclasses = 'UEM | STM | CTM | RTTM'
 
 
+
+class BaseABC:
+    @classmethod
+    def new(cls, d, **defaults):
+        # Example code:
+        # from meeteval.io.seglst import asseglst
+        # seglst = asseglst(d).map(lambda s: {**defaults, **s})
+        # ... (convert seglst to cls)
+        raise NotImplementedError(cls)
+
+    def to_seglst(self):
+        raise NotImplementedError()
+
+
 @dataclass(frozen=True)
 class BaseLine:
     @classmethod
     def parse(cls, line: str) -> 'Self':
         raise NotImplementedError(cls)
 
+    @classmethod
+    def from_dict(cls, segment: 'SegLstSegment') -> 'Self':
+        raise NotImplementedError(cls)
+
+    def to_seglst(self) -> 'SegLstSegment':
```

Comment:
Same issue as `from_seglst`, but I am nor sure, whether `to_dict` would be a good name.
It is too close to `dataclasses.asdict`.

Reply:
That's the reason why I didn't rename it yet. With the type annotation `def to_dict(self) -> 'SegLstSegment'` could be good enough, but it sounds too generic / too close to `dataclasses.asdict`

Reply:
Any suggestion for an alternative name? Or we may use the length `to_segLstSegment`

Reply:
Maybe `to_seg_lst_segment`? But that is so long. I think I prefer `to_dict` here

Reply:
I have a preference for the `to_seg_lst_segment`. While it is lengthy,
there are alternatives, that support autocomplete: `SegLstSegment.new(obj)`
The motivation for the existence of `to_seg_lst_segment` is mainly be used internally,
so a longer name is not an issue.

From the `to_dict` I would expect that the key names aren't changed.

Reply:
Okay, I renamed it to `to_seglst_segment` (to match `to_seglst`).